### ISSUE TITLE
Feature Request: Day and Night Mode

### ISSUE THREAD

**Issue by MAVRICK-1** [MAVRICK-1/e-commerce_website#1]
# Feature Request: Day and Night Mode
## Problem Description
The current website lacks a day and night mode feature, which can be inconvenient for users who prefer different display settings depending on the time of day or their personal preferences.

## Proposed Solution
Introduce a day and night mode feature on the website built with React.js to allow users to switch between light and dark modes easily. This feature would enhance the browsing experience by providing users with a more comfortable viewing environment.

## Alternatives Considered
One alternative solution could be to rely on browser extensions or third-party software to enable day and night mode. However, integrating this feature directly into the React.js application would offer a more seamless experience for users.

## Additional Context
- Adding a day and night mode feature can enhance accessibility and user experience.
- It can cater to a wider range of preferences and needs among users.
- Dark mode can reduce eye strain, especially during nighttime browsing sessions.

## Action Steps
1. Design and implement React components for toggling between day and night mode.
2. Utilize state management solutions like React Context or Redux to manage the current theme state.
3. Develop functionality to dynamically switch between light and dark themes based on user preference.
4. Test the feature within your React application, ensuring compatibility with different browsers and devices.

**Comment by Keshav-Chaudhary** [comment]
**I would like to take on the task of implementing this feature as I believe it aligns well with my skills and interests. Could you please assign this issue to me so that I can start working on it?**

Your guidance and feedback throughout the implementation process would be highly appreciated.

Thank you for considering my request

**Comment by MAVRICK-1** [comment]
@Keshav-Chaudhary Let's wait until tomorrow; mentors will be assigned to this project to guide you throughout the process, and points will also be allocated to respective issues, as the coding period will start tomorrow. This issue will be assigned to you on a first-come, first-served basis.

**Comment by MAVRICK-1** [comment]
@Keshav-Chaudhary as this issue have a lot to do , it will divided into 2 part both level 3

**Comment by ira-mane** [comment]
Helloo @MAVRICK-1  I would like to work on this issue .
So Should I create another issue for part two ? or would you assign it under the same issue ?

**Comment by MAVRICK-1** [comment]
@ira-mane If i didn't get any updates from @Keshav-Chaudhary it will assign to you

**Comment by ira-mane** [comment]
Ok thanks , But if I am not wrong you have commented the issue has to be divided into two parts as its big task ?

**Comment by MAVRICK-1** [comment]
@ira-mane okay i am assigning this to you , you can divide this issue in 3-4 parts as you link . and you can use material ui icon for the toggle button , create different issue for different parts

**Comment by MAVRICK-1** [comment]
@ira-mane each issue will be level 3 go ahead

**Comment by QQ07** [comment]
As you(@MAVRICK-1 ) mentioned about dividing this issue into 2 or 3 parts.. Can we(@ira-mane ) work together on this issue?
under GSSOC 24?

**Comment by ira-mane** [comment]
Okay sure why not @QQ07 How do you want the issue to be divided ? Let me know I will raise different issues in part and then you can be assigned too

**Comment by ira-mane** [comment]
Hello @MAVRICK-1  , it seems @QQ07 is not available as he has not replied to the text about how he want to proceed. So should I continue with the issue ? If yes ,I will part this issue into 3, 4 parts and will raise new issue and by 20th I will try to complete the work. Thanks.

**Comment by MAVRICK-1** [comment]
@ira-mane okay no issues , go ahead , and update your branch before starting

**Comment by MAVRICK-1** [comment]
@QQ07 try to make it in 3 parts

**Comment by ira-mane** [comment]
> @ira-mane okay no issues , go ahead , and update your branch before starting



> @ira-mane okay no issues , go ahead , and update your branch before starting

okay . Will tag you if I have any query .Thanks 👍
